File: Arrays/Easy/move_zeroes.py
def moveZeroes(nums):
    n = len(nums)
    firstzero = 0

    for i in range(n):
        if nums[i] != 0:
            while firstzero < n and nums[firstzero] != 0:
                firstzero += 1
            if firstzero < i:
                nums[firstzero], nums[i] = nums[i], nums[firstzero]


n = int(input())
v = list(map(int, input().split()))

moveZeroes(v)
print(v)

File: Arrays/Easy/move_zeroes.cpp
 #include<bits/stdc++.h>
 using namespace std;

 void moveZeroes(vector<int>& nums) {
          int n=nums.size();
          int firstzero=0;
          for(int i=0;i<n;i++){
                 if(nums[i]!=0){
                      while(firstzero<n && nums[firstzero]!=0){firstzero++;}
                      if(firstzero<i){
                      swap(nums[firstzero],nums[i]);}
                 }
          }
    }

int main(){
       int n;
       cin>>n;
       vector<int>v(n);
       for(int i=0;i<n;i++){
           cin>>v[i];
       }

       moveZeroes(v);
}